src/compiler/value/arithmetic.rs

```diff
@@ -73,7 +73,10 @@ impl VrlValueArithmetic for Value {
             Value::Integer(lhv) if rhs.is_float() => {
                 Value::from_f64_or_zero(lhv as f64 * rhs.try_float()?)
             }
-            Value::Integer(lhv) => (lhv * rhs.try_into_i64().map_err(|_| err())?).into(),
+            Value::Integer(lhv) => {
+                let rhv = rhs.try_into_i64().map_err(|_| err())?;
+                i64::wrapping_mul(lhv, rhv).into()
+            }
```

Comment:
Is overflow the general pattern we use arithmetic? Just figured it's worth writing that in https://github.com/vectordotdev/vector/blob/master/docs/DEVELOPING.md (or adding a similar file to VRL).

Reply:
I'm open to suggestions, but I figured it's worth at least making `debug` builds match what `release` builds currently do. Returning errors would require all multiplication to be fallible, which could be pretty painful to use. Another option might be to convert the result to an `f64` if it would overflow, but again that would change the type definition and probably make it pretty painful to use in practice. I can add a line to `DEVELOPING.md`.

Reply:
Although you bring up a good point, this should probably be consistent across all arithmetic. I'll modify those too. (addition, subtraction, etc.)